pkg/client/unversioned/clientcmd/config.go

```diff
@@ -75,8 +77,16 @@ func (o *PathOptions) GetLoadingPrecedence() []string {
 	if envVarFiles := o.GetEnvVarFiles(); len(envVarFiles) > 0 {
 		return envVarFiles
 	}
-
-	return []string{o.GlobalFile}
+	files, err := loadFileOrDir(o.GlobalFile)
+	// If GlobalFile doesn't exist, try GlobalFileDir
```

Comment:
Either/or seems confusing.  Was it explicitly requested this way?  I think I'd expect the file to have precedence over the directory, but loading both seems more predictable.